hooks/useAssetDetails.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from "react"
+
+import { OverviewContentSections, ResponsibilitiesContentSections } from "../components/AssetTabContent"
+import { HttpClient } from "../lib/HttpClient"
+
+type ErrorMessage= {
+  message: string
+}
+
+export const useAssetDetails = (assetId: string) => {
+  const [overviewData, setOverviewData] = useState<OverviewContentSections>()
+  const [responsibilitesData, setResponsibilitesData] = useState<ResponsibilitiesContentSections>()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<ErrorMessage | null>(null)
```

Comment:
Can be `useState<Error>()` since the `Error` interface has property `message`

Reply:
IRL discussion, landed on the string approach